testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -78,6 +78,20 @@
   end
 end
 
+When(/^I wait at most (\d+) seconds until I see "([^"]*)" text, refreshing the page$/) do |seconds, text|
+  next if has_text?(text, wait: 3)
+  repeat_until_timeout(message: "I still cannot see text '#{text}'", timeout: seconds.to_i) do
+    break if has_text?(text, wait: 3)
+    begin
+      accept_prompt do
+        execute_script 'window.location.reload()'
+      end
+    rescue Capybara::ModalNotFound
+      # ignored
+    end
+  end
+end
+
```

Comment:
I basically copied the step directly above and adjusted it.

Reply:
Use this one please:
`When(/^I wait until I see "([^"]*)" (text|regex), refreshing the page$/) do |text, type|`

Reply:
Another option, refactor the current step:
```
When(/^I wait at most "([^"]*)" seconds until I (see|do not see) "([^"]*)" text$/) do |seconds, action, text|
  is_shown = action == 'see'
  next if has_no_text?(text, wait: 3) == is_shown
  repeat_until_timeout(message: "I still see text '#{text}'", timeout: seconds.to_i) do
    break if has_no_text?(text, wait: 3) == is_shown
  end
end
```

Reply:
Ah, ok that uses the default timeout. Adjusted. Thank you!